### PR TITLE
[1.18] Mappings for `net.minecraft.client.gui`

### PR DESCRIPTION
**Pull request by geosskk**

### CODE REVIEW

File: data/net/minecraft/client/gui/Font.mapping [hunk 1]
```diff
@@ -182,6 +183,17 @@ CLASS net/minecraft/client/gui/Font
...
+		ARG 1 processor
+		ARG 2 x
+		ARG 3 y
+		ARG 4 color
+		ARG 5 isShadow
+		ARG 6 matrix
+		ARG 7 buffer
+		ARG 8 isTransparent
+		ARG 9 colorBackground
+		ARG 10 packedLight
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/Font.mapping]: These should follow the parameters names in the constructor (and method in the case for `colorBackground`/`backgroundColor`) for `StringRenderOutput`.
```suggestion
	METHOD renderText (Lnet/minecraft/util/FormattedCharSequence;FFIZLcom/mojang/math/Matrix4f;Lnet/minecraft/client/renderer/MultiBufferSource;ZII)F
		ARG 1 processor
		ARG 2 x
		ARG 3 y
		ARG 4 color
		ARG 5 dropShadow
		ARG 6 pose
		ARG 7 bufferSource
		ARG 8 seeThrough
		ARG 9 backgroundColor
		ARG 10 packedLightCoords
```

File: data/net/minecraft/client/gui/screens/multiplayer/WarningScreen.mapping [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CLASS net/minecraft/client/gui/screens/multiplayer/WarningScreen
+	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 title
+		ARG 2 content
+		ARG 3 checkbox
+		ARG 4 narration
+		ARG 5 lastScreen
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/multiplayer/WarningScreen.mapping]: Parameter names which are assigned directly to fields should be named the same as the field.
```suggestion
	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
		ARG 1 titleComponent
		ARG 2 content
		ARG 3 check
		ARG 4 narration
		ARG 5 previous
```

File: data/net/minecraft/client/gui/screens/recipebook/GhostRecipe.mapping [hunk 3]
```diff
@@ -8,6 +8,8 @@ CLASS net/minecraft/client/gui/screens/recipebook/GhostRecipe
 	METHOD render (Lcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/client/Minecraft;IIZF)V
 		ARG 1 poseStack
 		ARG 2 minecraft
+		ARG 3 leftPos
+		ARG 4 topPos
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/recipebook/GhostRecipe.mapping:12]: What would be better here: these names, or `x` and `y`?

File: data/net/minecraft/client/gui/screens/worldselection/CreateWorldScreen.mapping [hunk 4]
```diff
@@ -12,11 +12,21 @@ CLASS net/minecraft/client/gui/screens/worldselection/CreateWorldScreen
...
 		ARG 1 listener
+	METHOD copyBetweenDirs (Ljava/nio/file/Path;Ljava/nio/file/Path;Ljava/nio/file/Path;)V
+		ARG 0 fromDir
+		ARG 1 toDir
+		ARG 2 filePath
 	METHOD createFresh (Lnet/minecraft/client/gui/screens/Screen;)Lnet/minecraft/client/gui/screens/worldselection/CreateWorldScreen;
 		ARG 0 lastScreen
 	METHOD createFromExisting (Lnet/minecraft/client/gui/screens/Screen;Lnet/minecraft/server/WorldStem;Ljava/nio/file/Path;)Lnet/minecraft/client/gui/screens/worldselection/CreateWorldScreen;
 		ARG 0 lastScreen
+		ARG 1 worldstem
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/worldselection/CreateWorldScreen.mapping]: Camel-case to match the class name:
```suggestion
		ARG 1 worldStem
```

File: data/net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent.mapping [hunk 5]
```diff
@@ -31,5 +33,7 @@ CLASS net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent
 		ARG 1 visible
 	METHOD toString (Ljava/util/OptionalLong;)Ljava/lang/String;
 		ARG 0 seed
+	METHOD updateDataPacks (Lnet/minecraft/server/WorldStem;)V
+		ARG 1 worldstem
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent.mapping]: Camel-case to match the class name:
```suggestion
		ARG 1 worldStem
```

File: data/net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent.mapping [hunk 6]
```diff
@@ -20,6 +20,8 @@ CLASS net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent
 		ARG 3 preset
 	METHOD lambda$init$9 (Lnet/minecraft/client/Minecraft;Lnet/minecraft/client/gui/screens/worldselection/CreateWorldScreen;Lnet/minecraft/core/RegistryAccess$Writable;Lcom/mojang/serialization/Lifecycle;Lnet/minecraft/world/level/levelgen/WorldGenSettings;)V
 		ARG 5 settings
+	METHOD makeSettings (Z)Lnet/minecraft/world/level/levelgen/WorldGenSettings;
+		ARG 1 isHardcore
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent.mapping]: Could simplify the name to `hardcode`:
```suggestion
		ARG 1 hardcode
```

File: data/net/minecraft/client/gui/screens/worldselection/WorldSelectionList.mapping [hunk 7]
```diff
@@ -10,9 +10,15 @@ CLASS net/minecraft/client/gui/screens/worldselection/WorldSelectionList
 		ARG 9 worldSelectionList
 	METHOD moveSelection (Lnet/minecraft/client/gui/components/AbstractSelectionList$SelectionDirection;)V
 		ARG 1 ordering
+	METHOD refreshList (Ljava/util/function/Supplier;Z)V
+		ARG 1 supplier
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/worldselection/WorldSelectionList.mapping]: A more descriptive name would be something like `searchStringSupplier` or `filterStringSupplier`, as the supplied string is used to filter out any levels which do not contain the string as a substring of their level name or level ID.

File: data/net/minecraft/client/gui/screens/worldselection/WorldSelectionList.mapping [hunk 8]
```diff
@@ -10,9 +10,15 @@ CLASS net/minecraft/client/gui/screens/worldselection/WorldSelectionList
...
+		ARG 2 updateCache
 	METHOD setSelected (Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList$WorldListEntry;)V
 		ARG 1 entry
 	CLASS WorldListEntry
+		METHOD <init> (Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList;Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList;Lnet/minecraft/world/level/storage/LevelSummary;)V
+			ARG 2 worldSelectionList
+			ARG 3 levelSummary
```
Comment (sciwhiz12) [inline data/net/minecraft/client/gui/screens/worldselection/WorldSelectionList.mapping]: Parameter names which are assigned directly to fields should be named the same as the field.
```suggestion
		METHOD <init> (Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList;Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList;Lnet/minecraft/world/level/storage/LevelSummary;)V
			ARG 2 worldSelectionList
			ARG 3 summary
```

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

File: data/net/minecraft/client/gui/screens/recipebook/GhostRecipe.mapping (hunk 3 above)
Comment (geosskk) [reply data/net/minecraft/client/gui/screens/recipebook/GhostRecipe.mapping:12]: I'm not too sure, on the one hand it seems like `x` and `y` would be simpler, however every time this method is called the name of the fields that are used are `leftPos` and `topPos`.

For example, in `net/minecraft/client/gui/screens/inventory/CraftingScreen.java` the `render` method uses `this.leftPos` and `this.topPos`. _(This naming also occurs in `AbstractFurnaceScreen.java` and `InventoryScreen.java`)_
image

_The method `renderGhostRecipe` just calls the method `render` from `net/minecraft/client/gui/screens/recipebook/GhostRecipe.java`._
image

**Review by sciwhiz12 — APPROVED** [review]

**Review by alcatrazEscapee — APPROVED** [review]

**Pull request merged**